### PR TITLE
feat: lazy auth

### PR DESCRIPTION
**Pull request by NikolaMilosa**
This PR refactors the way we do proposals. Now the following rule applies:
> If the detection of neuron fails a fake neuron will be put in its place. That neuron doesn't exist on the mainnet and cannot submit proposals. 

This is done in order to allow the team to play around with different variations of commands without having to always say `--dry-run`. 

When the hsm is plugged in, or when the private key pem is specified the code will use that neuron.

### CODE REVIEW

File: rs/cli/src/auth.rs [hunk 1]
```diff
@@ -87,8 +87,29 @@ impl Neuron {
         Self::from_opts_and_req_inner(auth_opts, requirement, network, neuron_id).await
     }
 
-    pub async fn dry_run_fake_neuron(network: &Network) -> anyhow::Result<Self> {
-        Self::from_opts_and_req(AuthOpts::none(), AuthRequirement::Anonymous, network, None).await
+    fn ensure_fake_pem(name: &str) -> anyhow::Result<PathBuf> {
```
Comment (sasa-tomic) [inline rs/cli/src/auth.rs:90]:
```suggestion
    fn new_with_fake_pem(name: &str) -> anyhow::Result<PathBuf> {
```

File: rs/cli/src/auth.rs [hunk 2]
```diff
@@ -87,8 +87,29 @@ impl Neuron {
...
+        if !path.exists() {
+            std::fs::write(&path, "Some private key")?;
+        }
+        Ok(path)
+    }
+
+    pub fn dry_run_fake_neuron() -> anyhow::Result<Self> {
+        Ok(Self {
+            auth: Auth::Keyfile {
+                path: Self::ensure_fake_pem("test_neuron_1")?,
```
Comment (sasa-tomic) [inline rs/cli/src/auth.rs:108]:
```suggestion
                path: Self::new_with_fake_pem("test_neuron_1")?,
```

File: rs/cli/src/auth.rs (hunk 1 above)
Comment (nikolamilosasotex) [reply rs/cli/src/auth.rs:90]: But it just ensures that a file on that path exists. It doesn't return `Self`?

Comment (NikolaMilosa) [reply rs/cli/src/auth.rs:90]: But it just ensures that a path exists. It doesn't return `Self`?

Comment (sasa-tomic) [reply rs/cli/src/auth.rs:90]: ah right, my bad, sorry

File: rs/cli/src/auth.rs (hunk 2 above)
Comment (sasa-tomic) [reply rs/cli/src/auth.rs:108]: yep, sorry

**Review by sasa-tomic — APPROVED** [review]

**Pull request merged**